### ISSUE TITLE
Training Run: Ichigo-ASR-2502-en

### ISSUE THREAD

**Issue by Yip-Jia-Qi** [janhq/ichigo#209]
## Procedure:
Do a random init for codebook + freeze whisper encoder and decoder (medium)
Do not use whisperVQ checkpoint, init from scratch.
Codebook setting: 
        num_quantizers: 4 #whisperVQ used 1 only
        codebook_size: 512 #same as whisperVQ
        codebook_dim = 64 #same as whisperVQ
Use Phase 1 KL+CE only.  No need for CE-only phase 2.

## Question: 
-	Will we get better performance if we scale up the data?

## Training dataset

To be investigated

A useful point of reference is OWSM v3.1 low-restriction data:
https://www.wavlab.org/activities/2024/owsm/ 
          AMI: CC-BY-4.0
          Common Voice: CC0-1.0
          FLEURS: CC-BY-4.0
          KsponSpeech: MIT
          LibriSpeech: CC-BY-4.0
          Multilingual LibriSpeech: CC-BY-4.0
          VCTK: CC-BY-4.0

See how many of these datasets we can get quickly and start training with minimal data engineering. Hopefully there are at least 500-1000 hours of data we can use out of the box. Before we start, we do a quick estimation of how long it will take.

## Evaluation dataset
Use test-clean and test-other as the test set
Earnings22 dataset: https://paperswithcode.com/dataset/earnings-22

**Comment by tuanlda78202** [comment]
Dataset 

* [AMI](https://huggingface.co/datasets/edinburghcstr/ami?row=36) (CC-BY-4.0)
* [CommonVoice 17.0](https://huggingface.co/datasets/mozilla-foundation/common_voice_17_0) (CC0-1.0)

**Comment by Yip-Jia-Qi** [comment]
Can we prioritise downloading the test sets for this dataset, and add this to our list of standardised benchmarks?

**Comment by tuanlda78202** [comment]
The download process is currently running, I will benchmark 3 models on test-set soon!

**Comment by tuanlda78202** [comment]
@Yip-Jia-Qi Updated benchmark on LargeScaleASR on https://github.com/janhq/ichigo-factory/issues/30#issuecomment-2665917299

**Comment by tuanlda78202** [comment]
# Experiments

* Started training `ichigo-asr-2502-en` on 8xA6000s
* Trained on `clean` subset of LargeScaleASR, 3.94M samples (over 13k audio hours)
* Config
  * `num_quantizer` = 1
  * `vq_codes` = 1024
  * `whisper_model` = `large-v3`
* [Tracking log](https://wandb.ai/tuanlda78202/ichigo-asr)

**Comment by Yip-Jia-Qi** [comment]
Can you update the experimental tracking table with the following hyperparams?

dim
num_quantizers
codebook_size
codebook_dim

Then also write down the rest of the VQ params that we will be keeping constant.

**Comment by tuanlda78202** [comment]
* Exp
  ```bash
  whisper_model="medium"
  num_quant=4
  codebook_size=512
  codebook_dim=512
  ```
* Benchmark (Canary-1b)[https://huggingface.co/nvidia/canary-1b]

**Comment by Yip-Jia-Qi** [comment]
Do the following experiments

| Base_model | Total Epochs | `num_quantizers` | `codebook_size ` | `dim` |`codebook_dim` | Prompt | WER | 
| --- | --- | --- | --- | --- | --- | --- | --- | 
medium | 7 | 4 | 512 | 80 | 64 | 
medium | 7 | 4 | 512 | 80 | 128 |
medium | 7 | 8 | 512 | 80 | 64 |

**Comment by tuanlda78202** [comment]
# Experiments w/ `large-v3`

* 250225

| ID | Epoch | `whisper_name`  | dim | `num_quantizers` | `codebook_size ` | `codebook_dim` | Prompt | WER | 
| --- | --- | --- |--- |--- | --- | --- | --- | --- | 
| 250225 | 1 | medium | 8 | 4 | 512  | 64 | old_prompt |14.53 |

**Comment by Yip-Jia-Qi** [comment]
For future reference:
`dim` refers to the `dim` parameter of the RVQ module https://github.com/janhq/ichigo-factory/blob/ea61c391d91f33684183debcf84e993bf5c2ea44/ichigo-asr/models/vq_transformer.py#L109
